src/components/query-viewer/query-viewer.tsx

```diff
@@ -0,0 +1,332 @@
+  /**
+   * Include max and min
+   */
+  @Prop() includeMaxMin: boolean;
+  /**
+   * Theme for showing the graph
+   */
+  @Prop() theme: string;
+  /**
+   * Object type -> Query or DataProvider
+   */
+  @Prop() objectType: string;
+  /**
+   * True if it is external query
+   */
+  @Prop() isExternalQuery: boolean;
+  /**
+   * Allowing elements order to change
+   */
+  @Prop() allowElementsOrderChange: boolean;
+  /**
+   * If type== PivotTable or Table, if true will shrink the table
+   */
+  @Prop() autoResize: boolean;
+  /**
+   * If autoResize, in here select the type, Width, height, or both
+   */
+  @Prop() autoResizeType: string;
+  /**
+   * Type of font
+   */
+  @Prop() fontFamily: string;
+  /**
+   * Font size
+   */
+  @Prop() fontSize: number;
+  /**
+   * Font Color
+   */
+  @Prop() fontColor: string;
+  /**
+   * Auto refresh group
+   */
+  @Prop() autoRefreshGroup: string;
+  /**
+   * Allowing or not Comlumn sort
+   */
+  @Prop() disableColumnSort: boolean;
+  /**
+   * Allow selection
+   */
+  @Prop() allowSelection: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XML
+   */
+  @Prop() exportToXML: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to HTML
+   */
+  @Prop() exportToHTML: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XLS
+   */
+  @Prop() exportToXLS: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to XLSX
+   */
+  @Prop() exportToXLSX: boolean;
+  /**
+   * If type== PivotTable or Table allow to export to PDF
+   */
+  @Prop() exportToPDF: boolean;
+  /**
+   * Title of the QueryViewer
+   */
+  @Prop() queryTitle: string;
+
+  private generateForm() {
+    const form = document.createElement("form");
+    form.action = this.baseUrl + this.mapServices[this.env];
+    form.method = "POST";
+    form.target = "query_viewer";
+    form.innerHTML = this.postData();
+
+    document.body.append(form);
+    form.submit();
+  }
+
+  private postData(): string {
+    let postBody = "";
+    for (const key in Object(this)) {
+      if (
+        Object(this)[key] != undefined &&
+        !this.propsNotToPost.includes(key)
+      ) {
+        postBody +=
+          '<input type="hidden" name="' +
+          key +
+          '" value="' +
+          Object(this)[key] +
+          '" />';
+      }
+    }
+    postBody +=
+      '<input type="hidden" name="Width" value="' + this.getWidth() + '" />';
+    postBody +=
+      '<input type="hidden" name="Height" value="' + this.getHeight() + '" />';
+    postBody +=
+      '<input type="hidden" name="Elements" value="' +
+      this.getElements() +
+      '" />';
+    postBody +=
+      '<input type="hidden" name="Parameters" value="' +
+      this.getParameters() +
+      '" />';
+    return postBody;
+  }
+
+  private getParameters(): string {
+    let parametersValue = "";
+    const parameters = Array.from(
+      document.getElementsByTagName("gx-query-viewer-parameter")
```

Comment:
What happens if I programmatically add a new `gx-query-viewer-parameter` child element?

Reply:
It isn't supposed ti happen this.
All parameters should come when it renders or it wouldn't work.

Reply:
I'm thinking of the following scenario:

The query viewer is rendered with an initial set of parameters (that were added by the Angular template as `gx-query-viewer-parameter` elements). 
Then, in a GeneXus user event, the set of parameters is changed, which will make the template to render a new set of `gx-query-viewer-parameter`.

With the current approach, we wouldn't be able to render the new parameters. Maybe it's not the case with parameters but what about the other child elements?

Reply:
If I understood correct how it will work, it totally make sense if this happens this way.

For example, on Start Event:
receives parameters, CountryName: Spain, CityName: Madrid
Then another event happens and send:
CountryName: Italy, CityName: Rome

We want to replace the previous parameters with the new ones.

And same logic would work for the other child elements